### ISSUE TITLE
options with hypens are truncated

### ISSUE THREAD

**Issue by djerius** [polettix/App-Easer#9]
If an option name contains a hyphen, the option name is truncated at the hyphen.

The syntax is supported by Getopt::Long:

```
% perl -MGetopt::Long -E 'GetOptions( q{gti-filter-file=s}, \$foo ); say $foo' -- --gti-filter-file 33
33
```

But not in `App::Easer`

```
use App::Easer::V2 'run';

exit run( {
        help    => "Merge, evaluate, and other stuff",
        execute => sub {
            my $app = shift;
            use Data::Dumper;
            say Dumper( $app->config_hash );
        },
        options => [ {
                getopt  => "gti-filter-file",
                default => 1
            },
            { getopt => 'help|h' },
        ],
        default_child => '-self',
    },
    $0,
    @ARGV
);
```
which results in
```
$VAR1 = {
          'gti' => 1
        };
```

**Comment by polettix** [comment]
The problem is not with the option's name but with how the option name is inferred at the time of setting the default value. You can see that if you call your program passing option `--gti-filter-file`.

This is fixed in dd8e113 and will be part of the next release, thanks for taking the time to report the bug!

> By the way, take note that the *default* options for Getopt::Long are set via `getopt_config`, which by default is `gnu_getopt` in App::Easer; see it [here](https://github.com/polettix/App-Easer/blob/4c4a875063156b987a5d61d6396e19715e51f5d7/lib/App/Easer/V2.pm#L250). You can override this by explicitly setting it in the "usual" manner (either with a key in the app's definition hash, or by overriding the method in a derived class, see [here](https://github.com/polettix/App-Easer/blob/dd8e1134dc9663ee63e72e59542f4ca52398de49/lib/App/Easer/V2.pod#application-high-level-view)).

**Comment by polettix** [comment]
[Release v2.008 is out](https://metacpan.org/release/POLETTIX/App-Easer-2.008)
